importlib_metadata/__init__.py

```diff
@@ -957,7 +1102,37 @@ def version(distribution_name):
     return distribution(distribution_name).version
 
 
-def entry_points(**params) -> Union[EntryPoints, SelectableGroups]:
+@overload
+def entry_points() -> SelectableGroups:
+    ...  # pragma: no cover
+
+
+@overload
+def entry_points(*, name: str, **params: str) -> EntryPoints:
+    ...  # pragma: no cover
+
+
+@overload
+def entry_points(*, value: str, **params: str) -> EntryPoints:
+    ...  # pragma: no cover
+
+
+@overload
+def entry_points(*, group: str, **params: str) -> EntryPoints:
+    ...  # pragma: no cover
+
+
+@overload
+def entry_points(*, module: str, **params: str) -> EntryPoints:
```

Comment:
Although it's possible, I'd not expect callers to use `value=`, `module=`, or `attr=` and I'd not want to declare it as supported unless we chose specifically to support it. I mainly expect users to use `name=` and `group=`.